ext/fetch/fs_fetch_handler.rs

```diff
@@ -25,25 +38,48 @@ pub struct FsFetchHandler;
 impl FetchHandler for FsFetchHandler {
   fn fetch_file(
     &self,
-    _state: &mut OpState,
+    state: &mut OpState,
     url: &Url,
   ) -> (CancelableResponseFuture, Option<Rc<CancelHandle>>) {
+    let mut access_check = sync_permission_check::<PermissionsContainer>(
+      state.borrow_mut(),
+      "fetch()",
+    );
     let cancel_handle = CancelHandle::new_rc();
-    let path_result = url.to_file_path();
+    let path = match url.to_file_path() {
+      Ok(path) => path,
+      Err(_) => {
+        let fut = async move { Err::<_, _>(()) };
+        return (
+          fut
+            .map_err(move |_| super::FetchError::NetworkError)
+            .or_cancel(&cancel_handle)
+            .boxed_local(),
+          Some(cancel_handle),
+        );
+      }
+    };
+    let file = open_with_access_check(
+      OpenOptions {
+        read: true,
+        ..Default::default()
+      },
+      &path,
+      Some(&mut access_check),
+    );
```

Comment:
Should we do this in a blocking task?

Reply:
Does `open()` ever block? Either way we can't move it to the future because of the lifetime of `access_check`.

Reply:
Uh actually its easy lol, gimme a min